lib/pbench/server/api/resources/query_apis/datasets/__init__.py

```diff
@@ -129,12 +129,17 @@ def get_index(self, dataset: Dataset, root_index_name: AnyStr) -> AnyStr:
         try:
             index_map = Metadata.getvalue(dataset=dataset, key=Metadata.INDEX_MAP)
         except MetadataError as exc:
-            self.logger.error("{}", str(exc))
-            raise APIAbort(HTTPStatus.INTERNAL_SERVER_ERROR)
+            self.logger.warning("{}", str(exc))
+            raise APIAbort(
+                HTTPStatus.INTERNAL_SERVER_ERROR,
+                f"Required metadata missing: {str(exc)!r}",
+            )
 
         if index_map is None:
-            self.logger.error("Index map metadata has no value")
-            raise APIAbort(HTTPStatus.INTERNAL_SERVER_ERROR)
+            self.logger.warning("Index map metadata has no value")
+            raise APIAbort(
+                HTTPStatus.INTERNAL_SERVER_ERROR, "Required metadata missing"
+            )
```

Comment:
Why are Internal Server Errors (now) being reported as "warnings" instead of "errors", here?  (Aren't they "errors" in most places?)

Reply:
Interesting point. My logic was that with #3080 we're handling this case better; but that's a mental slip because that has nothing to do with this method, which isn't in an "expected failure" context. Although it arguably should be, anyway, since it just means "you made this call too early". I probably should lock all the Elasticsearch queries on `Indexed` state and not cry about "missing required metadata". *That's* definitely not for this PR, but the change to `warning` was probably unjustified. (Sigh.)